resources/packs/k8s/ingress.go

```diff
@@ -225,3 +232,63 @@ func (k *mqlK8sIngressservicebackend) id() (string, error) {
 func (k *mqlK8sIngressresourceref) id() (string, error) {
 	return k.Id()
 }
+
+func getCertificates(ingress *networkingv1.Ingress, getSecrets func() ([]interface{}, error)) ([]interface{}, error) {
+	certificates := []interface{}{}
+	if len(ingress.Spec.TLS) > 0 {
+		// This returns ALL Secrets found in the cluster!
+		secretsInterface, err := getSecrets()
+		if err != nil {
+			return nil, fmt.Errorf("failed to fetch Secrets referenced in Ingress: %s", err)
+		}
+
+		secrets := []*mqlK8sSecret{}
+		for _, secInterface := range secretsInterface {
```

Comment:
I think you can optimize this a bit:
- You can only need use secrets that are in the same namespace as your ingress. This means that you can already discard all secrets that are in a different namespace.
- Instead of adding the `mqlK8sSecret` structs to a slice, you can add them to a map with the key of the map being the name (there won't be duplicate names if you keep secrets from 1 ns as mentioned in the first point)